pkg/tkctl/readable/printers.go

```diff
@@ -167,26 +168,29 @@ func printPod(pod *v1.Pod, options printers.PrintOptions) ([]metav1beta1.TableRo
 	if options.Wide {
 		row.Cells = append(row.Cells, columns.PodIP, columns.NodeName)
 	}
-	componentKind := pod.Labels[label.ComponentLabelKey]
-	switch componentKind {
-	case label.TiKVLabelVal:
-		extraInfoColumn := extraTikvDataColumn(pod)
-		row.Cells = append(row.Cells, extraInfoColumn.StoreId)
-		break
-	default:
-		break
-	}
 	return []metav1beta1.TableRow{row}, nil
 }
 
+// add more columns for tikv column
 func printTikvList(tikvList *alias.TikvList, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
-	podList := apiv1.PodList(*tikvList)
-	return printPodList(&podList, options)
-}
-
-func printTikv(tikv *alias.Tikv, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
-	pod := apiv1.Pod(*tikv)
-	return printPod(&pod, options)
+	podList := tikvList.PodList
+	metaTableRows, err := printPodList(podList, options)
+	if err != nil {
+		return nil, err
+	}
+	for id, row := range metaTableRows {
+		podName := row.Cells[podNameIndex].(string)
+		var storeId string
+		for _, pod := range tikvList.PodList.Items {
+			if podName == pod.Name {
+				storeId = pod.Labels[label.StoreIDLabelKey]
+				break
+			}
+		}
+		row.Cells = append(row.Cells, storeId, tikvList.TikvStatus.Stores[storeId].State)
```

Comment:
It is better to fall back to `<none>` than panic if there is any error accessing the state